move/axelar_gateway/sources/gateway.move

```diff
@@ -328,41 +342,19 @@ public fun take_approved_message(
 // -----------------
 
 fun peel_messages(message_data: vector<u8>): vector<Message> {
-    let mut bcs = bcs::new(message_data);
-
-    let mut messages = vector::empty<Message>();
-    let mut len = bcs.peel_vec_length();
-
-    while (len > 0) {
-        messages.push_back(message::peel(&mut bcs));
-
-        len = len - 1;
-    };
-
-    assert!(bcs.into_remainder_bytes().length() == 0, ERemainingData);
-    assert!(messages.length() > 0, EInvalidLength);
-
-    messages
-}
-
-fun peel_weighted_signers(weighted_signers_data: vector<u8>): WeightedSigners {
-    let mut bcs = bcs::new(weighted_signers_data);
-
-    let weighted_signers = weighted_signers::peel(&mut bcs);
-
-    assert!(bcs.into_remainder_bytes().length() == 0, ERemainingData);
-
-    weighted_signers
-}
-
-fun peel_proof(proof_data: vector<u8>): Proof {
-    let mut bcs = bcs::new(proof_data);
-
-    let proof = proof::peel(&mut bcs);
-
-    assert!(bcs.into_remainder_bytes().length() == 0, ERemainingData);
-
-    proof
+    peel_data!(
+        message_data,
+        |bcs| {
+            let mut messages = vector::empty<Message>();
+            let mut len = bcs.peel_vec_length();
+            while (len > 0) {
+                messages.push_back(message::peel(bcs));
+                len = len - 1;
+            };
```

Comment:
```suggestion
            let messages = vector::tabulate(bcs.peel_vec_length(), |_| message::peel(bcs));
```
use macro funs to remove all while loops
https://blog.sui.io/move-2024-macros-beta/